src/EVM/Expr.hs

```diff
@@ -954,6 +954,9 @@ simplify e = if (mapExpr go e == e)
 
     -- redundant CopySlice
     go (CopySlice (Lit 0x0) (Lit 0x0) (Lit 0x0) _ dst) = dst
+    go (CopySlice (Lit 0) (Lit 0) (BufLength k1)
+      (CopySlice (Lit 0) (Lit 0) (BufLength k2) k3 _)
+      (ConcreteBuf "")) | k1 == k2 && k2 == k3 = k1
```

Comment:
So, I have a suggestion to split this into two, more general rules.

First is to remove redundant nested `CopySlice`:
`(CopySlice srcOff2 dstOff2 size2 (CopySlice srcOff1 dstOff1 size1 src _) dst) | size1 == size2 && srcOff2 == dstOff1 = CopySlice srcOff1 dstOff2 size1 src dst)`

The second one the CopySlice into empty buffer:
`(CopySlice (Lit 0) (Lit 0) (BufLength src) src (ConcreteBuf "")) = src`

Your rule should be a consequence of these two.

Reply:
Thanks! I have now added this :) Looking great!